classic_ml/tabular_dataset.py

```diff
@@ -0,0 +1,58 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright (c) 2022, Ampere Computing LLC
+
+import numpy as np
+import pandas as pd
+
+class TabularDataset:
+    """
+    A class providing facilities for preprocessing and postprocessing of ImageNet validation dataset.
+    """
+
+    def __init__(self, file_path, batch_size=8, task='classification'):       
+        self.__batch_size = batch_size
+        self.__y_true = []
+        self.__y_pred = []
+        self.__k = 0
+        self.task = task
+        df = pd.read_csv(file_path)
+        self.X = df.iloc[:,:-1].values
+        self.y = df.iloc[:,-1].values
+        self.available_instances = float('inf')
```

Comment:
is the number really infinite? this should be a finite value

Reply:
Because we generate duplicate samples ideally we can have large num of samples.